test/linear_combination.jl

```diff
@@ -0,0 +1,202 @@
+@testset "linear combination" begin
+
+    lc = LinearCombination([1, -1])
+    @test lc isa Transform
+
+    @testset "Vector" begin
+        x = [1, 2]
+        expected = [-1]
+
+        @testset "all inds" begin
+            @test Transforms.apply(x, lc) == expected
+            @test lc(x) == expected
+        end
+
+        @testset "dims not supported" begin
+            @test_throws  MethodError Transforms.apply(x, lc; dims=1)
+        end
+
+        @testset "dimension mismatch" begin
+            x = [1, 2, 3]
+            @test_throws DimensionMismatch Transforms.apply(x, lc)
+        end
+
+        @testset "specified inds" begin
+            x = [1, 2, 3]
+            inds = [2, 3]
+            expected = [-1]
+
+            @test Transforms.apply(x, lc; inds=inds) == expected
+            @test lc(x; inds=inds) == expected
+        end
+    end
+
+    @testset "Matrix" begin
+        M = [1 1; 2 2; 3 5]
+        expected = [0, 0, -2]
+
+        @testset "all inds" begin
+            @test Transforms.apply(M, lc) == expected
+            @test lc(M) == expected
+        end
+
+        # TODO: Colon() is not supported by eachslice(A, dims=:), do we care?
```

Comment:
What about this? I believe the Power tests use Colon() for dims which is why this question came up

Reply:
Interesting.... I'm not sure how to interpret a linear combination (with a vector of coefficients) taken across all dimensions...

Do you apply the LC to one dimension first, then the other? and get a scalar as a result?

In the above cases what we're doing is reducing the dimension of the input by 1, so Matrix-> Vector, Vector->Scalar.
To do that you need to specify the dimension across which you're reducing the elements.

So I don't think (for vector coefficients) we can support `Colon()` since it's undefined which dimension you're taking the sum over.

Does this make sense?